Use https by default when downloading images

jellonek opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mirantis/virtlet/246)
<!-- Reviewable:end -->

pigmej commented:
I think it breaks demo. Needs to be fixed ;)

---

Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[cmd/virtlet/virtlet.go, line 2 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMaLD3JmoHGGWhcvOz:-KhMaLD4eWqD6qtjTHHW:b-gwf6xq) ([raw file](https://github.com/mirantis/virtlet/blob/6c46f0ff67c6e5fca4d9abf0263bc45a592ceb20/cmd/virtlet/virtlet.go#L2)):*
> -2017 

Could we pls NOT do it with the refactorings etc?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246#-:-KhMauAfNc3OijUrQCl2:b-3vryt)*
<!-- Sent from Reviewable.io -->

jellonek commented:
Updated demo and verified it on my dev env - works as before changes.

---

Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[cmd/virtlet/virtlet.go, line 2 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMaLD3JmoHGGWhcvOz:-KhMgEcaNVF_ROtyyMJP:b-hfmjyr) ([raw file](https://github.com/mirantis/virtlet/blob/6c46f0ff67c6e5fca4d9abf0263bc45a592ceb20/cmd/virtlet/virtlet.go#L2)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary><blockquote>

> -2017 

Could we pls NOT do it with the refactorings etc?
</blockquote></details>

Moved to separate commit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246#-:-KhMgIzqHsd-_AX3pxTa:bv31xi1)*
<!-- Sent from Reviewable.io -->

jellonek commented:
Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[cmd/virtlet/virtlet.go, line 2 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMaLD3JmoHGGWhcvOz:-KhMiW5lhX6DYiFBSidm:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/6c46f0ff67c6e5fca4d9abf0263bc45a592ceb20/cmd/virtlet/virtlet.go#L2)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Moved to separate commit.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246)*
<!-- Sent from Reviewable.io -->

ivan4th commented:
Review status: 0 of 13 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/libvirttools/image.go, line 53 at r2](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMyDRFWM4ra7y-Ae4V:-KhMyDRFWM4ra7y-Ae4W:bg3niiv) ([raw file](https://github.com/mirantis/virtlet/blob/04c0a2cb5b5e63be9db6af8aefdeb88ea6448aad/pkg/libvirttools/image.go#L53)):*
> ```Go
> 	}
> 	if protocol == "" {
> 		protocol = "https"
> ```

I'd use a constant named `DEFAULT_PROTOCOL` or something

---

*[pkg/utils/download.go, line 28 at r2](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMyJ100t4haCoNskRo:-KhMyJ100t4haCoNskRp:b-dal6xj) ([raw file](https://github.com/mirantis/virtlet/blob/04c0a2cb5b5e63be9db6af8aefdeb88ea6448aad/pkg/utils/download.go#L28)):*
> ```Go
> )
> 
> // DownloadFile saves pointed by protocol/location raw data under fileName in /tmp
> ```

// DownloadFile downloads a file via an URL constructed as 'protocol://location' and saves it under the specified fileName in /tmp

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246)*
<!-- Sent from Reviewable.io -->

jellonek commented on deploy/demo.sh:
```diff
@@ -124,7 +124,7 @@ function demo::vm-ready {
 function demo::start-virtlet {
   demo::step "Deploying Virtlet DaemonSet"
-  "${kubectl}" create -f "${BASE_LOCATION}/deploy/virtlet-ds.yaml"
+  VIRTLET_DOWNLOAD_PROTOCOL=http "${kubectl}" create -f "${BASE_LOCATION}/deploy/virtlet-ds.yaml"
```
Hmmm... Hold on. This will not work as I wanted.

pigmej commented:
Review status: 0 of 13 files reviewed at latest revision, 3 unresolved discussions.

---

*[deploy/demo.sh, line 127 at r3](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMblRMKN5aKpcB2Opz-r3-127:-KhNYcgm_xyO4BR4NDmT:b-u4xknq) ([raw file](https://github.com/mirantis/virtlet/blob/ba19e4cddf35ea8f0252f6ff2b7d88b8bbc02256/deploy/demo.sh#L127)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Hmmm... Hold on. This will not work as I wanted.
</blockquote></details>

rly that much complexity? Not to mention that it requires `jq` on kube-master (which may not be always true, right?) `sed -i` would do the work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246)*
<!-- Sent from Reviewable.io -->

pigmej commented:
Review status: 0 of 13 files reviewed at latest revision, 3 unresolved discussions.

---

*[deploy/demo.sh, line 127 at r3](https://reviewable.io:443/reviews/mirantis/virtlet/246#-KhMblRMKN5aKpcB2Opz-r3-127:-KhNaUNhO_amu9pAE8e5:b-xcdkxr) ([raw file](https://github.com/mirantis/virtlet/blob/ba19e4cddf35ea8f0252f6ff2b7d88b8bbc02256/deploy/demo.sh#L127)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary><blockquote>

rly that much complexity? Not to mention that it requires `jq` on kube-master (which may not be always true, right?) `sed -i` would do the work.
</blockquote></details>

Anyway, let's use it. it's already used in demo

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246)*
<!-- Sent from Reviewable.io -->

ivan4th commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 13 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/246#-:-KhXHFIqSg4vvnvCp8-f:bnfp4nl)*
<!-- Sent from Reviewable.io -->
